kubeval/kubeval.go

```diff
@@ -90,64 +55,64 @@ func detectLineBreak(haystack []byte) string {
 	return "\n"
 }
 
-func determineSchema(kind string, apiVersion string) string {
+func determineSchema(kind, apiVersion string, config *Config) string {
 	// We have both the upstream Kubernetes schemas and the OpenShift schemas available
-	// the tool can toggle between then using the --openshift boolean flag and here we
+	// the tool can toggle between then using the config.Openshift boolean flag and here we
 	// use that to select which repository to get the schema from
 
-	// Set a default Version to make usage as a library easier
-	if Version == "" {
-		Version = "master"
-	}
 	// Most of the directories which store the schemas are prefixed with a v so as to
 	// match the tagging in the Kubernetes repository, apart from master.
-	normalisedVersion := Version
-	if Version != "master" {
+	normalisedVersion := config.KubernetesVersion
+	if normalisedVersion != "master" {
 		normalisedVersion = "v" + normalisedVersion
 	}
 
-	// Check Viper for environment variable support first.
-	// Then check for an override in SchemaLocation
-	// Finally settle on the default value
-	baseURLFromEnv := viper.GetString("schema_location")
-	var baseURL string
-	if baseURLFromEnv != "" {
-		baseURL = baseURLFromEnv
-	} else if SchemaLocation == "" {
-		if OpenShift {
-			baseURL = OpenShiftSchemaLocation
-		} else {
-			baseURL = DefaultSchemaLocation
-		}
-	} else {
-		baseURL = SchemaLocation
-	}
-
-	var strictSuffix string
-	if Strict {
+	strictSuffix := ""
+	if config.Strict {
 		strictSuffix = "-strict"
-	} else {
-		strictSuffix = ""
 	}
 
-	var kindSuffix string
-
 	groupParts := strings.Split(apiVersion, "/")
 	versionParts := strings.Split(groupParts[0], ".")
 
-	if OpenShift {
-		kindSuffix = ""
-	} else {
+	kindSuffix := ""
+	if !config.OpenShift {
 		if len(groupParts) == 1 {
 			kindSuffix = "-" + strings.ToLower(versionParts[0])
 		} else {
 			kindSuffix = fmt.Sprintf("-%s-%s", strings.ToLower(versionParts[0]), strings.ToLower(groupParts[1]))
 		}
 	}
 
+	baseURL := determineBaseURL(config)
 	return fmt.Sprintf("%s/%s-standalone%s/%s%s.json", baseURL, normalisedVersion, strictSuffix, strings.ToLower(kind), kindSuffix)
 }
 
+func determineBaseURL(config *Config) string {
+	// Order of precendence:
+	// 1. If --openshift is passed, return the openshift schema location
```

Comment:
I'm pretty sure this doesn't change the existing behavior. The old code relied on viper, which bound the environment variable to the flag, and [viper gives the flag a higher precedence](https://github.com/spf13/viper#why-viper). This behavior is demonstrated in the series of acceptance tests added in https://github.com/instrumenta/kubeval/pull/157.

Further, I think this order of precedence is closer to the norm; we can see [here](https://kubernetes.io/docs/concepts/configuration/organize-cluster-access-kubeconfig/#merging-kubeconfig-files) that kubectl follows this pattern with the `kubeconfig` flag, and [here](https://helm.sh/docs/helm/#options) that helm follows it with the `home` flag.

The `openshift` flag can go wherever we want it, but I think it makes the most sense as the highest precedent, since it is the most specific of the configurations.